### ISSUE TITLE
question about class weights

### ISSUE THREAD

**Issue by emoebel** [ELEKTRONN/ELEKTRONN#6]
Hello,

I have a question about class_weights. I understand that 1 is default value, <1 means less important classe, >1 means more important class, and that the values are normalized internally. But is there a way to put this weights in relation to the frequency of appearance of each class? 

Could you maybe share the formula of how the weights are normalized and how they are integrated in the cost function?

Cheers,
Lazyleaf

**Comment by xeray** [comment]
Hi Lazyleaf,

yes you can relate the the frequency and the choosing of weights. Choose the weight of each class in order to equalise the prevalences of the classes. E.g. if index 0 is your background class and  index 1 your "event" class and you have 90% BG and 10% events (as measured by pixel count or instance count depending on your context), you can choose the weights as (0.2, 1.8) (so the second weight is 9x the first weight and the sum is 2 - as in the balanced case where the weights would be (1,1) ).
Atm I removed the automatic normalisation of the weights, because this is actually not neccessary since the the scale of the weights just scales the gradients whose scales are controlled by the learning rate anyway.

Note that such a forceful "balancing" of rare events might produce many false positives and needs a re-calibration of the optimal classifier threshold after training. You might also try a compromise with just a little bit more weight on the rare classes.

**Comment by emoebel** [comment]
Thanks for the explanation, class weight are more clear to me now :-)
